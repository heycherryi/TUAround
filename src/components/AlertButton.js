import React from 'react';
import PropTypes from 'prop-types';

function AlertButton({ alerts, buttonAction }) {
  if (alerts.length > 0) {
    return (
      <button className="btn btn-sm btn-primary bg-dark" onClick={buttonAction}>
          <span className="badge bg-light text-dark">
            {alerts.length}
          </span>&nbsp;
          Service Alerts
      </button>
    );
  }
}

AlertButton.propTypes = {
  alerts: PropTypes.array.isRequired,
  buttonAction: PropTypes.func.isRequired,
};

AlertButton.defaultProps = {
  alerts: [],
};

export default AlertButton;
