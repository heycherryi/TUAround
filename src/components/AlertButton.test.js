/* globals test, expect */

import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import AlertButton from './AlertButton';

const alertsFixture = require('../fixtures/alerts.json');

test('renders AlertButton', () => {
  const { container } = render(
    <Router>
      <AlertButton alerts={alertsFixture} buttonAction={() => undefined} />
    </Router>,
  );
  expect(screen.getByText('Service Alerts')).toBeInTheDocument();
  expect(screen.getByText('8')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

test('does not render AlertButton when no alerts', () => {
  const { container } = render(
    <Router>
      <AlertButton alerts={[]} buttonAction={() => undefined} />
    </Router>,
  );
  expect(container).toMatchSnapshot();
});
