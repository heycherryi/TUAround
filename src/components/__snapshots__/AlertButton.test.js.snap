// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render AlertButton when no alerts 1`] = `<div />`;

exports[`renders AlertButton 1`] = `
<div>
  <button
    class="btn btn-sm btn-primary bg-dark"
  >
    <span
      class="badge bg-light text-dark"
    >
      8
    </span>
    Â  Service Alerts
  </button>
</div>
`;
